### ISSUE TITLE
Missing dl3 header keywords

### ISSUE THREAD

**Issue by LukasNickel** [cta-observatory/cta-lstchain#1220]
Currently lstchain does not add the header keywords related to the earth location to the dl3 files:
https://gamma-astro-data-formats.readthedocs.io/en/latest/general/coordinates.html?highlight=geolon#earth-location

This is not required for most standard analyses, but if it is, gammapy is unable to perform the coordinate transformations and will fail with somewhat cryptic error messages.

Since it is not difficult to add and is the same for all runs (already defined here: https://github.com/cta-observatory/cta-lstchain/blob/97716ecf0cf676b96d0936cc3839253da1e4bce3/lstchain/reco/utils.py#L54), it should be included in my opinion.

**Comment by morcuended** [comment]
I agree

**Comment by maxnoe** [comment]
> , is that these are locations of array centres and not of the single telescope, like LST-1.

For single-telescope arrays, as is the case right now, the location of the telescope is obviously identical to the array center. We do not have an issue here.

Besides, this is used for coordinate transformations between AltAz and ICRS, so differences of ~100m on the ground in the earth location don't really matter for us.

>  (either following FITS standard or existing IACT FITS convention)

Again, no issue here, we are writing out data in  GADF format, so we need to follow GADF (GEOLON, GEOLAT, ALTITUDE).
See https://gamma-astro-data-formats.readthedocs.io/en/v0.3/general/coordinates.html#earth-location

> I don't know if ctapipe has a solution for this.

I don't see how ctapipe would be involved here, just write the LST location into the headers mentioned above.

**Comment by chaimain** [comment]
I was just wondering in case, we might have plans to follow more closely to the CTA North Array. But ok, this can be updated for the DL3 files.

**Comment by maxnoe** [comment]
> I was just wondering in case, we might have plans to follow more closely to the CTA North Array.

We need to fill in the most appropriate earth location for the coordinate transformation. This means it should be the same location we used for transforming the reconstructed from horizontal (alt/az) values into ICRS (ra/dec). This is currently the coordinate defined in `reco/utils.py`.

**Comment by maxnoe** [comment]
>I don't see how ctapipe would be involved here.

Ah, one way that ctapipe *is* involved: since version 0.16, the `SubarrayDescription` has a `reference_location` member, which is filled by the `EventSource` to be compatible with the simulations, which are using the mirror-area-weighted average of LST-1 and MAGIC:

https://github.com/cta-observatory/ctapipe_io_lst/blob/068fc1f0da7e3afaa2b0b1664398fed6eef08652/src/ctapipe_io_lst/constants.py#L37-L42

Maybe we should just use that for everything (coordinate transformations and then putting it into DL3).

**Comment by chaimain** [comment]
The `ctapipe_io_lst.constants.LST1_LOCATION` 
```
#: location of lst-1 as `~astropy.coordinates.EarthLocation`
#: Taken from Abelardo's Coordinates of LST-1 & MAGIC presentation
#: https://redmine.cta-observatory.org/attachments/65827
LST1_LOCATION = EarthLocation(
    lon=-17.89149701 * u.deg,
    lat=28.76152611 * u.deg,
    # height of central pin + distance from pin to elevation axis
    height=2184 * u.m + 15.883 * u.m
)
```

is different from `lstchain.reco.utils.location`
```
# position of the LST1
location = EarthLocation.from_geodetic(-17.89139 * u.deg, 28.76139 * u.deg, 2184 * u.m)
```

Will this difference be insignificant for the coordinate transformations we have been doing so far?

**Comment by chaimain** [comment]
Also, I had added the location information to the POINTING HDU Header only https://github.com/cta-observatory/cta-lstchain/blob/main/lstchain/high_level/hdu_table.py#L494-L505 , and that too not following the GADF convention. I can fix this as well

**Comment by maxnoe** [comment]
@chaimain The difference between the two is 24m, so no, it won't affect the coordinate transformations in any significant way.

I don't know where the current definition in lstchain comes from, the one in ctapipe_io_lst is from the presentation by Abelardo linked in the docstring .

**Comment by chaimain** [comment]
The values were fixed from #123 using Google Earth
